### PR TITLE
feat(squad): hats-first sponsor deploy with ext ala carte and ungated safe

### LINKED ISSUE #89: chore: adopt redeployed Ext sponsor factory and split deployer from addressOwner
## Summary

Ala carte **Squad sponsor (Ext)** should let a member **deploy with their Default (funded) wallet** while setting the clone **`addressOwner`** to their **squad-assigned roster EVM**. The current on-chain factory always sets `addressOwner = msg.sender`, and the app currently requires the squad roster key to pay deploy gas — coupling funder and owner.

Contract work to add an explicit `addressOwner` parameter and redeploy Sepolia lives in the sponsor repo. This issue tracks the **pacto-app follow-up** after that redeploy: new protocol addresses, Rust/TS deploy wiring, and Ext deploy UX.

**Blocked by (contract + redeploy):** [covenant-gov/pacto-squad-sponsor#6](https://github.com/covenant-gov/pacto-squad-sponsor/issues/6)

## Dependency

Do **not** merge app changes until [pacto-squad-sponsor#6](https://github.com/covenant-gov/pacto-squad-sponsor/issues/6) publishes:

- New `SquadSponsorFactory` address (Sepolia)
- Updated `createSquadSponsorExt(squadId, addressOwner)` ABI
- Paymaster / entryPoint addresses if bundled in the same redeploy manifest

### 1. Protocol address book

- [ ] Update [`src/lib/evm/pacto-protocol-addresses.json`](https://github.com/covenant-gov/pacto-app/blob/main/src/lib/evm/pacto-protocol-addresses.json) with new Sepolia `squadSponsor.factory` (and related fields if changed)
- [ ] Verify `src-tauri` `pacto_chain_config` reads the new factory at compile/runtime
- [ ] Note old factory in `docs/wallet/PROTOCOL_ADDRESS_BOOK.md` only if migration context is needed (greenfield: replace in place)

### 2. Rust deploy path

Files: `src-tauri/src/evm/squad_sponsor_deploy.rs`, `src-tauri/src/evm/contracts/pacto_sponsor/mod.rs`

- [ ] Extend `deploy_squad_sponsor_for_parent` to pass **`addressOwner`** (squad roster EVM) into `createSquadSponsorExt`
- [ ] Support **`signerWallet: default`** for Ext ala carte deploy (gas + `msg.value` deposit from Default signer)
- [ ] Keep `signerWallet: squad` when deployer and owner are the same roster key
- [ ] Persist `addressOwner` in `provider_payload` / announce card (already partially read via `addressOwnerCall`)

### 3. Frontend Ext deploy wizard

File: `src/components/parent/governance/DeploySquadSponsorExtModal.svelte`

- [ ] Default payer = **Default signer** (funded DM wallet)
- [ ] Display squad-assigned address as **`addressOwner`** (not payer) — clarify copy: “You deploy and fund; squad roster address owns the allowlist”
- [ ] Optional: dual wallet picker (same pattern as deposit form / combined gov+sponsor wizard)
- [ ] Pass `addressOwner` + `signerWallet` through `deploySquadSponsorForParent`

### 4. API + tests

- [ ] `src/lib/governance/api.ts` — new `addressOwner` param on Ext deploy invoke
- [ ] `api.test.ts` + Rust deploy encoding tests for new calldata shape
- [ ] Update `docs/wallet/OPERATOR_SMOKE.md` Ext ala carte steps (Default-funded deploy, roster owner)

## Acceptance criteria

- [ ] On Sepolia, Ext deploy from launchpad with **Default wallet funded** + **squad roster as owner** succeeds without funding the roster key
- [ ] On-chain `addressOwner()` on new clone equals squad-assigned roster EVM; deployer ≠ owner when wallets differ
- [ ] `pacto-protocol-addresses.json` points at factory from [pacto-squad-sponsor#6](https://github.com/covenant-gov/pacto-squad-sponsor/issues/6)
- [ ] Existing squads deployed on **old** factory are not broken (no forced migration); only **new** Ext deploys use new factory

## Out of scope

- ERC-4337 sponsored governance writes — [#87](https://github.com/covenant-gov/pacto-app/issues/87) / [pacto-squad-sponsor#5](https://github.com/covenant-gov/pacto-squad-sponsor/issues/5)
- Mainnet redeploy (Sepolia first)

## Test plan

- Sepolia smoke: Default-funded Ext deploy; verify owner + pool deposit on explorer
- Negative: missing roster binding blocks owner field; insufficient Default balance blocks deploy

### PR DESCRIPTION
**Pull request by daopunk**
## Summary

Make squad gas sponsorship deployable and operable, and adopt the redeployed Ext factory so Default can fund deploy while the squad roster EVM owns the allowlist.

Before: Ext sponsor ownership was tied to `msg.sender`, so Default-funded deploy either failed product intent or left ACL on the wrong key. Safe deploy was gated behind sponsor/gov infra. Hats-linked sponsor lived outside the main gov wizard.

After: combined Pacto Gov + hats sponsor wizard (optional crew bootstrap); ala carte Ext from the launchpad; Default may pay deploy/deposit while captain/crew hats and Ext `addressOwner` stay on roster EVMs; Safe deploy/import are independent of sponsor/gov; Sepolia address book points at the factory from pacto-squad-sponsor#7 (`createSquadSponsorExt(squadId, addressOwner)`).

closes #89

## Changes

- Combined gov + hats-linked sponsor deploy path with optional crew bootstrap and dual payer (Default vs squad roster)
- Advanced Ext sponsor deploy: roster as `addressOwner`, payer may be Default; new Sepolia factory/paymaster in `pacto-protocol-addresses.json`
- Launchpad routes Treasury/Governance deploy CTAs; Crew tab shows Ext sponsor-member status; hat singularity on deploy
- Ungate Safe deploy from sponsor/gov prerequisites

## Follow-up

**Not in this PR:** [#87](https://github.com/covenant-gov/pacto-app/issues/87) — route hat-gated governance writes through the squad sponsor ERC-4337 paymaster when the roster key has 0 ETH (bootstrap crew, QM/treasury/mutiny mutations).

Contract/client blockers for #87 are cleared via [pacto-squad-sponsor#8](https://github.com/covenant-gov/pacto-squad-sponsor/pull/8) (desktop integration guide + golden vectors). That work needs a new UserOp / EIP-7702 submission path and is intentionally a **separate follow-up PR** so this change stays focused on deploy/deposit, Ext factory adopt, and launchpad UX.

Until #87 lands, captains with an empty roster key still need ETH (or must wait for sponsored UserOps) for on-chain gov writes even when the sponsor pool is funded.

## Test plan

- [ ] Sepolia: Ext ala carte with funded Default + roster `addressOwner`; confirm on-chain owner ≠ deployer when wallets differ
- [ ] Combined gov + hats sponsor (+ optional bootstrap) from launchpad
- [ ] Deposit with Default payer; hats remain on roster EVMs
- [ ] Deploy Safe without sponsor/gov infra
- [ ] `pnpm test` / relevant `cargo test` for governance + sponsor deploy helpers

---

[Compound Engineering](https://github.com/EveryInc/compound-engineering-plugin)

### CODE REVIEW

File: src-tauri/src/evm/nave_pirata_deploy.rs [hunk 1]
```diff
@@ -141,6 +155,7 @@ pub async fn deploy_nave_pirata_for_parent<R: Runtime>(
     captain: String,
```
Comment (logicminds) [inline src-tauri/src/evm/nave_pirata_deploy.rs:144]: **[P0] Unauthorized Pacto Gov deploy + arbitrary captain**
Why: `deploy_nave_pirata_for_parent` accepts a user-supplied `captain` address and passes it to the Nave Pirata factory, which mints the captain hat to that address. With no membership check for the caller and no roster validation for the captain, any user can deploy governance for any parent and claim the captain hat, seizing control of the squad's treasury and governance.
Suggested fix: Add a `require_parent_member(app, pid)` check and validate that the `captain` address is one of the roster EVM addresses for the parent (from `db::list_squad_member_evm`). The caller should be a member of the parent, and the captain should be a member's shared EVM, not an arbitrary user-supplied address.
Reviewers: security | Confidence: 100

File: src-tauri/src/evm/squad_sponsor_deploy.rs [hunk 2]
```diff
@@ -9,8 +9,13 @@ use serde::Serialize;
...
+use super::contracts::pacto_sponsor::ISquadSponsorExt::addressOwnerCall;
+use super::contracts::pacto_sponsor::ISquadSponsorFactory::{
+    createSquadSponsorCall, createSquadSponsorExtCall,
+};
+use super::gov_read::parse_top_hat_id;
 use super::pacto_chain_config;
+use super::rpc::call::eth_call_decode;
 use super::rpc::{
     connect_read_provider, connect_signing_provider, contract_call_request, send_and_confirm,
     wallet_err_json, wallet_err_json_with_tx_hash,
```
Comment (logicminds) [inline src-tauri/src/evm/squad_sponsor_deploy.rs:21]: **[P0] Default signer bypasses roster binding for deploys**
Why: The new `signer_wallet` option lets the default (DM) wallet pay for deploys. `require_treasury_signing_allowed` only validates that the active wallet is phrase-derived and squad-purpose; it does not verify membership in the target `parent_id`. An attacker with any phrase-derived squad wallet can therefore deploy infra for squads they do not belong to.
Suggested fix: When `signer_wallet == 'default'`, still require a roster binding for the parent before deploying. The payer being the user's default wallet does not prove they are a member of the squad. Add a parent-membership check that is independent of the chosen signer wallet.
Reviewers: security | Confidence: 100

---

**1. [P0] Unauthorized Ext sponsor deploy + owner takeover**
Why: `deploy_squad_sponsor_for_parent` calls the factory with `createSquadSponsorExt`, which deploys a clone whose owner is the transaction sender. Because the command does not verify the caller is a member of `parent_id`, any account with a squad-purpose wallet can front-run a real squad, deploy its sponsor, and become the owner, blocking the legitimate sponsor and controlling the allowlist.
Suggested fix: Add a `require_parent_member(app, pid)` preflight that checks the current account has a row in `squad_member_evm` or `squad_member_evm_account` for the parent before calling `createSquadSponsorExt`. The Ext variant makes `msg.sender` the owner, so membership gating is essential to prevent any user from seizing a squad's sponsor.
Reviewers: security | Confidence: 100

**2. [P1] Rust sponsor deploy has no transaction flow tests**
Suggested fix: Add unit tests for parse_required_deposit_wei, parse_signer_wallet, and the failure paths of deploy_squad_sponsor_for_parent (line 74) and deploy_squad_sponsor_hats_for_parent (line 196): empty parent, unsupported network, zero deposit, already deployed, registry mismatch, and malformed signer input. Add mocked-provider tests for receipt parsing and DB persistence.
Reviewers: testing | Confidence: 100

**3. [P1] deploy_squad_sponsor_for_parent and deploy_squad_sponsor_hats_for_parent are near-duplicates**
Suggested fix: Extract a shared `deploy_squad_sponsor_with_calldata(app, network, parent_id, initial_deposit_wei, signer_wallet, factory, calldata, variant_label, |sponsor, variant| -> serde_json::Value)` helper that performs the common validation, signer loading, RPC setup, send_and_confirm, and persistence; have both commands build their variant-specific calldata and payload extras and delegate to it.
Reviewers: maintainability | Confidence: 100

**4. [P1] Ext sponsor deploy lacks pre-deploy duplicate guard**
Why: A user or compromised frontend can attempt to deploy a second Ext sponsor for a parent that already has one. The first transaction may already have consumed the squadId mapping in the factory, causing the second to revert and waste gas, or—depending on factory semantics—could create duplicate state.
Suggested fix: In deploy_squad_sponsor_for_parent, add the same preflight used by deploy_squad_sponsor_hats_for_parent: db::parent_has_sponsor_infra check and an on-chain read_squad_record check before sending createSquadSponsorExtCall.
Reviewers: adversarial, correctness | Confidence: 100

File: src-tauri/src/evm/squad_admin_deploy.rs [hunk 3]
```diff
@@ -133,7 +132,6 @@ pub async fn deploy_squad_admin_for_parent<R: Runtime>(
             None,
```
Comment (logicminds) [inline src-tauri/src/evm/squad_admin_deploy.rs:134]: **[P1] Squad Admin deploy not restricted to captain**
Why: `deploy_squad_admin_for_parent` is meant to be a captain/executor management tool. The current code only requires a roster-bound treasury signer, so any member with a roster binding can deploy a Squad Admin and, in the `ext_standalone` variant, set themselves or any address as owner.
Suggested fix: Add a membership check for the parent and, when Pacto Gov exists, require a captain capability. For the `ext_standalone` variant, restrict the `owner` parameter to the caller's roster address or a member's roster address; do not allow arbitrary addresses.
Reviewers: security | Confidence: 100

File: src-tauri/src/evm/squad_sponsor_deploy.rs [hunk 4]
```diff
@@ -154,6 +163,167 @@ pub async fn deploy_squad_sponsor_for_parent<R: Runtime>(
...
+        paymaster_address: format!("{:#x}", paymaster),
+        variant: variant_str,
+        provider_payload: payload,
+        infra_row_id,
+    })
+}
+
+/// Deploy a hat-first SquadSponsor clone linked to an existing Nave Pirata top hat.
+#[tauri::command]
+pub async fn deploy_squad_sponsor_hats_for_parent<R: Runtime>(
```
Comment (logicminds) [inline src-tauri/src/evm/squad_sponsor_deploy.rs:203]: **[P1] Hats sponsor deploy lacks captain authorization**
Why: Deploying a hats-linked sponsor is a privileged governance action that should be restricted to the captain. The current code only checks that the signer is a roster-bound treasury signer, allowing any member with a roster EVM to deploy the sponsor, set the top hat link, and block the intended sponsor configuration.
Suggested fix: Gate `deploy_squad_sponsor_hats_for_parent` behind a captain-level authorization. Since Pacto Gov already exists, use `require_capability(app, pid, GovCapability::CaptainResign)` or a new `DeploySponsor` capability, or require that the caller's roster address matches the captain address stored in the Pacto Gov infra.
Reviewers: security | Confidence: 100

File: src-tauri/src/evm/squad_sponsor_deposit.rs [hunk 5]
```diff
@@ -34,6 +39,7 @@ pub async fn deposit_squad_sponsor<R: Runtime>(
     parent_id: String,
```
Comment (logicminds) [inline src-tauri/src/evm/squad_sponsor_deposit.rs:40]: **[P1] Rust sponsor deposit has no unit tests**
Suggested fix: Add tests for deposit_squad_sponsor covering empty parent, zero/invalid amount, unsupported network, signer selection (default vs squad), explicit sponsor address vs registry lookup, sponsor registry mismatch, and read_sponsor_pool after deposit.
Reviewers: testing | Confidence: 100

File: src-tauri/src/evm/squad_sponsor_ext.rs [hunk 6]
```diff
@@ -0,0 +1,308 @@
...
+    pub chain_id: u64,
+    pub parent_id: String,
+    pub sponsor_address: String,
+    pub address_owner: String,
+    pub hats_wired: bool,
+    pub member_permits: Vec<SquadSponsorExtMemberPermit>,
+}
+
+#[tauri::command]
+pub async fn get_squad_sponsor_ext_status(
```
Comment (logicminds) [inline src-tauri/src/evm/squad_sponsor_ext.rs:88]: **[P1] Rust sponsor ext status and guards are barely tested**
Suggested fix: Add tests for resolve_sponsor_for_parent, get_squad_sponsor_ext_status (empty/invalid/duplicate member addresses, hats_wired), and squad_sponsor_set_permitted_address (hats_wired guard, non-owner rejection, zero-address member).
Reviewers: testing | Confidence: 100

File: src/components/parent/ParentDashboard.svelte [hunk 7]
```diff
@@ -562,38 +639,56 @@ import { TREASURY_SAFE_UI_CAP, governanceTreasurySafeForParent, vaultTreasurySaf
   }
```
Comment (logicminds) [inline src/components/parent/ParentDashboard.svelte:629]: **[P1] Import Safe still gated on squad sponsor**
Suggested fix: Remove the `requireSponsorForInfra` wrapper in `openSetSafe` and open the import modal when `parentId` is present, mirroring `openDeploySafe` at line 641.
Reviewers: correctness | Confidence: 100

File: src/components/parent/governance/DeployPactoGovAndSponsorModal.svelte [hunk 8]
```diff
@@ -0,0 +1,705 @@
...
+    const opt = captainMemberOptions.find((o) => o.address.toLowerCase() === key);
+    return { address: addr, label: opt?.label?.trim() || '' };
+  });
+
+  function onDepositInput(e: Event) {
+    const el = e.currentTarget as HTMLInputElement;
+    initialDepositEth = normalizeLeadingDotDecimalInput(el.value);
+  }
+
+  function executeDeploy() {
```
Comment (logicminds) [inline src/components/parent/governance/DeployPactoGovAndSponsorModal.svelte:264]: **[P1] Combined wizard closes modal before multi-step job finishes**
Why: The wizard runs Nave Pirata deploy, then hats sponsor, then optional bootstrap in a background job. executeDeploy calls onClose() immediately after startPactoGovAndSponsorDeploy returns ok. With no in-flight guard, the user can reopen the launchpad and click Deploy again while the first job is still between steps, risking a second Nave Pirata transaction and cascading gas losses.
Suggested fix: Keep the modal open (or replace it with a persistent progress panel) and disable the Deploy button until the background job reports success or failure. Maintain a per-parent in-flight flag so the launchpad cannot start a second deploy while one is pending.
Reviewers: adversarial | Confidence: 100

File: src/components/parent/governance/DeployPactoGovAndSponsorModal.svelte [hunk 9]
```diff
@@ -0,0 +1,705 @@
...
+    if (ok) {
+      onClose();
+    }
+  }
+
+  $: deployDisabled =
+    !squadNetwork ||
+    resolvingAddresses ||
+    depositExceedsBalance ||
+    (!sponsorOnly && (captainMemberOptions.length === 0 || !captainAddress)) ||
```
Comment (logicminds) [inline src/components/parent/governance/DeployPactoGovAndSponsorModal.svelte:364]: **[P1] Modal dismisses before background job reports failure**
Why: executeDeploy closes the modal immediately when startPactoGovAndSponsorDeploy returns true, but the real work runs in runOnChainInBackground. If the job later fails, onError sets deployError on a component the user can no longer see.
Suggested fix: Keep the modal open as a progress/failure pane until onSuccess fires; only close on successful completion. Move the close call into the onSuccess handler instead of immediately after startPactoGovAndSponsorDeploy returns true.
Reviewers: reliability | Confidence: 100

File: src/components/parent/governance/DeployPactoGovAndSponsorModal.svelte [hunk 10]
```diff
@@ -0,0 +1,705 @@
...
+    </p>
+  {/if}
+
+  {#if deployError}
+    <p class="input-error" role="alert">{deployError}</p>
+  {/if}
+
+  <div class="modal-actions">
+    <button type="button" class="btn-secondary" on:click={onClose}>Cancel</button>
+    <button type="button" class="btn-primary" disabled={deployDisabled} on:click={executeDeploy}>
```
Comment (logicminds) [inline src/components/parent/governance/DeployPactoGovAndSponsorModal.svelte:588]: **[P1] Deploy button lacks in-flight guard**
Suggested fix: Add a `deploying` boolean flag. Set it to true at the start of `executeDeploy` (line ~216) and return early if already true; include it in `deployDisabled` (line ~307); clear it in the `onError`/`onComplete` callbacks passed to `startPactoGovAndSponsorDeploy`/`startHatsSponsorOnlyDeploy`. This prevents double-click or rapid reopen+click from spawning concurrent background on-chain jobs.
Reviewers: julik-frontend-races | Confidence: 100

File: src/components/parent/governance/DeploySquadSponsorExtModal.svelte [hunk 11]
```diff
@@ -357,12 +247,12 @@
...
+        {squadBalance.symbol} on {deployNetwork} so this wallet can pay gas.
       </p>
-    {:else if selectedAddress && depositWei !== null}
+    {:else if squadCanonical && depositWei !== null}
       <p class="sponsor-deploy-hint">
-        Depositing from {shortAddress(selectedAddress)}.
+        Depositing from {shortAddress(squadCanonical)}.
       </p>
     {/if}
   </div>
```
Comment (logicminds) [inline src/components/parent/governance/DeploySquadSponsorExtModal.svelte:389]: **[P1] Ext sponsor deploy button unguarded**
Suggested fix: Add a `deploying` boolean flag. Set it to true in `confirmDeploy` (line ~151) and return early if already true; include it in `canDeploy`; clear it in the `runOnChainInBackground` `onSuccess`/`onError` callbacks. This prevents double-click from starting two `deploySquadSponsorForParent` background jobs.
Reviewers: julik-frontend-races | Confidence: 100

File: src/lib/evm/on-chain-background.ts [hunk 12]
```diff
@@ -72,7 +72,10 @@ export function runOnChainInBackground<T>(opts: RunOnChainInBackgroundOpts<T>):
     } catch (e) {
```
Comment (logicminds) [inline src/lib/evm/on-chain-background.ts:72]: **[P1] on-chain-background.ts has no unit tests**
Suggested fix: Add tests for runOnChainInBackground and waitForOnChainConfirmationInBackground covering startedToast, onSuccess, onError, error parsing, and callback timing.
Reviewers: testing | Confidence: 100

File: src/lib/governance/start-pacto-gov-and-sponsor-deploy.ts [hunk 13]
```diff
@@ -0,0 +1,392 @@
...
+    return { bootstrapped: true };
+  } catch (e) {
+    const bootstrapError = getInvokeErrorMessage(e, 'Crew bootstrap failed.');
+    console.error('[bootstrap]', e, bootstrapError);
+    return { bootstrapped: false, bootstrapError };
+  }
+}
+
+/** Sequential Nave Pirata → hats sponsor → optional bootstrapCrew. */
+export function startPactoGovAndSponsorDeploy(params: {
```
Comment (logicminds) [inline src/lib/governance/start-pacto-gov-and-sponsor-deploy.ts:140]: **[P1] Combined gov+sponsor deploy orchestrator is untested**
Suggested fix: Add tests for startPactoGovAndSponsorDeploy (line 136) and startHatsSponsorOnlyDeploy (line 270) that validate input rejection, runOnChainInBackground invocation, gov→sponsor→bootstrap ordering, bootstrap soft-fail handling, and callback firing.
Reviewers: testing | Confidence: 100

File: src/lib/governance/start-pacto-gov-and-sponsor-deploy.ts [hunk 14]
```diff
@@ -0,0 +1,392 @@
...
+  const crewCandidates = params.bootstrapCrew
+    ? bootstrapCrewCandidates(params.memberOptions, captain)
+    : [];
+  if (params.bootstrapCrew && crewCandidates.length === 0) {
+    const message =
+      'Bootstrap needs at least one shared EVM that is not the captain. Uncheck bootstrap or wait for more members.';
+    if (params.onReject) params.onReject(message);
+    else showToast(message);
+    return false;
+  }
```
Comment (logicminds) [inline src/lib/governance/start-pacto-gov-and-sponsor-deploy.ts:197]: **[P1] Combined deploy leaves partial infra on sponsor failure**
Why: If the second transaction fails after the first has already persisted a Pacto Gov infra row, the squad ends up with live governance but no gas sponsor. The user must discover the launchpad recovery path on their own.
Suggested fix: Wrap the gov → sponsor → bootstrap sequence in a saga or add a compensation step. If the sponsor deploy fails after gov succeeds, surface a 'Finish sponsor' recovery action in the error toast/modal rather than treating the whole flow as a failure.
Reviewers: reliability | Confidence: 100

File: src/components/parent/governance/DeployPactoGovAndSponsorModal.svelte [hunk 15]
```diff
@@ -0,0 +1,705 @@
...
+    preferFundedDefaultWhenRosterEmpty();
+  }
+
+  onMount(() => {
+    void refreshSigners();
+  });
+
+  $: if (!resolvingAddresses && !captainAddress && captainMemberOptions.length > 0) {
+    pickDefaultCaptain(canonicalAddress(squadSignerAddress));
+  }
```
Comment (logicminds) [inline src/components/parent/governance/DeployPactoGovAndSponsorModal.svelte:222]: **[P1] No timeout on signer and balance lookups**
Why: refreshSigners awaits both the roster address and native balances without any deadline. If the backend RPC or Tauri bridge hangs, the modal stays in resolvingAddresses forever and the Deploy button never enables.
Suggested fix: Wrap getEvmNativeBalance and resolveSquadRosterEvmAddress in Promise.race with a timeout (e.g., 15s) and set resolvingAddresses = false plus an error state on timeout.
Reviewers: reliability | Confidence: 75

File: src/components/parent/governance/LaunchpadModal.svelte [hunk 16]
```diff
@@ -35,70 +38,100 @@
...
           disabled={channelBlocked}
           on:click={() => {
             onClose();
-            onDeploySponsor();
+            onDeployGovAndSponsor();
           }}
         >
           Deploy squad sponsor
         </button>
+      {:else if hasSponsor && !hasPactoGov}
```
Comment (logicminds) [inline src/components/parent/governance/LaunchpadModal.svelte:73]: **[P1] Launchpad lacks Pacto Gov-only recovery path**
Suggested fix: Add a Pacto Gov-only advanced card (or rewire the primary card) when `hasSponsor && !hasPactoGov`, and route it to the existing `DeployPactoGovModal` wiring in `ParentDashboardModals.svelte` instead of leaving `openPactoGovDeploy`/`onDeployPactoGov` dead.
Reviewers: correctness | Confidence: 75

**Review by logicminds — CHANGES_REQUESTED** [review]
Automated code review (ce-code-review) on PR #86.

**Verdict: Not ready**

- Scope: PR head `d74b42b8` vs base `630a1014`
- Files changed: 36
- Total findings: 58
- P0: 3, P1: 17, P2: 31, P3: 7
- Actionable findings: 51
- Pre-existing findings: 3
- Reviewers: correctness, testing, maintainability, project-standards, security, performance, api-contract, reliability, adversarial, julik-frontend-races, ce-agent-native-reviewer, ce-learnings-researcher
- Run ID: 20260716-104210-2a64499a

This review was generated from a local worktree checkout (`pacto-app/.worktrees/pr-86-review`) and the ce-code-review multi-agent pipeline. Inline comments below cover all P0 and P1 findings.

**Review by logicminds — APPROVED** [review]
I pushed another branch with fixes for this branch and we can merge after this one.
